test/integration/executor/create_asset_test.cpp

```diff
@@ -0,0 +1,114 @@
+
+#include "integration/executor/executor_fixture.hpp"
+
+#include <gtest/gtest.h>
+#include "common/result.hpp"
+#include "framework/common_constants.hpp"
+#include "framework/crypto_literals.hpp"
+#include "integration/executor/command_permission_test.hpp"
+#include "integration/executor/executor_fixture_param_provider.hpp"
+#include "module/shared_model/mock_objects_factories/mock_command_factory.hpp"
+#include "module/shared_model/mock_objects_factories/mock_query_factory.hpp"
+
+using namespace common_constants;
+using namespace executor_testing;
+using namespace framework::expected;
+using namespace shared_model::interface::types;
+
+using shared_model::interface::permissions::Grantable;
+using shared_model::interface::permissions::Role;
+
+static const AssetNameType kAssetName{"new_asset"};
+constexpr PrecisionType kAssetPrecision(1);
+
+const AssetIdType &getNewId() {
+  static const AssetIdType kNewId{kAssetName + "#" + kDomain};
+  return kAssetId;
+}
+
+class CreateAssetTest : public ExecutorTestBase {
+ public:
+
+  void checkNoSuchAsset(
+      const boost::optional<AssetIdType> asset_id = boost::none) {
+    auto asset_val = asset_id.value_or(getNewId());
+    checkQueryError<shared_model::interface::NoAssetErrorResponse>(
+        getItf().executeQuery(
+            *getItf().getMockQueryFactory()->constructCreateAsset(
+                asset_val)),
+        0);
+  }
+
+  iroha::ametsuchi::CommandResult createAsset(
+      const AccountIdType &issuer,
+      const AssetNameType &target_name = kAssetName,
+      const DomainIdType &target_domain = kDomain,
+      constexpr PrecisionType  &precision = kAssetPrecision,
+      bool validation_enabled = true) {
+    return getItf().executeCommandAsAccount(
+        *getItf().getMockCommandFactory()->constructCreateAsset(
+            target_name, target_domain, precision),
+        issuer,
+        validation_enabled);
+  }
+//
+  iroha::ametsuchi::CommandResult createDefaultAsset(
+      const AccountIdType &issuer, bool validation_enabled = true) {
+    return createAsset(
+        issuer, kAssetName, kDomain, kAssetPrecision, validation_enabled);
+  }
+};
+
+using CreateAssetBasicTest = BasicExecutorTest<CreateAssetTest>;
+
+/**
+ *  given a user with all related permissions
+ *  when executes CreateAsset command with nonexistent domain
+ *  then the command does not succeed and the asset is not added

```

Comment:
this is a syntax for some tool that generates descriptions..
```suggestion
 *  @given a user with all related permissions
 *  @when executes CreateAsset command with nonexistent domain
 *  @then the command does not succeed and the asset is not added
```